userspace/libsinsp/socket_handler.h

```diff
@@ -1561,14 +1553,21 @@ class socket_data_handler
 			{
 				*(parser_data->m_msg_completed) = true;
 			}
-			else { throw sinsp_exception("Socket handler: parser msg complete null."); }
+			else { throw sinsp_exception("Socket handler (http_msg_completed_callback): parser data m_msg_completed is null."); }
+			if(parser_data->m_http_response)
+			{
+				*(parser_data->m_http_response) = parser->status_code;
+			}
+			else { throw sinsp_exception("Socket handler (http_msg_completed_callback): parser data m_http_response is null."); }
 		}
-		else { throw sinsp_exception("Socket handler: parser or data null."); }
+		else { throw sinsp_exception("Socket handler (http_msg_completed_callback): parser or data null."); }
 		return 0;
 	}
 
 	void init_http_parser()
```

Comment:
When the exception is thrown in `socket_handler`, it will be caught in [socket_collector](https://github.com/draios/sysdig/blob/dev/userspace/libsinsp/socket_collector.h#L297) and all `socket_handler`s will be removed from the `socket_collector`; when a `k8s_handler` finds it was kicked out, it will [delete](https://github.com/draios/sysdig/blob/dev/userspace/libsinsp/k8s_handler.cpp#L275) its `socket_handler`, whose destructor will call [cleanup()](https://github.com/draios/sysdig/blob/dev/userspace/libsinsp/socket_handler.h#L81), which will [free](https://github.com/draios/sysdig/blob/dev/userspace/libsinsp/socket_handler.h#L1517) the http parser.

We may be "painting with a too broad brush" here and this could be optimized by kicking out only the offending handler from the collector, but attention should be paid to whether the handler is or not critical (eg. we can't consistently proceed without `namespace` or `node` handlers, but can survive without eg. daemonset handler, see [explanation](https://github.com/draios/sysdig/blob/dev/userspace/libsinsp/k8s_handler.h#L180) and [component classification](https://github.com/draios/sysdig/blob/dev/userspace/libsinsp/k8s_component.cpp#L245)).